app/models/saved_claim/veteran_readiness_employment_claim.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'sentry_logging'
+
+class SavedClaim::VeteranReadinessEmploymentClaim < SavedClaim
+  include SentryLogging
+  FORM = '28-1900'
+  class VREError < StandardError; end
+
+  def add_claimant_info(user)
+    return if form.blank?
+
+    updated_form = parsed_form
+
+    updated_form['veteranInformation'] = {
+      'fullName' => {
+        'first' => user.first_name,
+        'middle' => user.middle_name || '',
+        'last' => user.last_name
+      },
+      'ssn' => user.ssn,
+      'VAFileNumber' => veteran_va_file_number(user),
+      'pid' => user.participant_id,
+      'edipi' => user.edipi,
+      'vet360ID' => user.vet360_id,
+      'dob' => user.birth_date
+    }
+
+    update(form: updated_form.to_json)
+  end
+
+  def send_to_vre
+    conn = Faraday.new(url: Settings.vre.base_url)
+
+    response = conn.post do |req|
+      req.url Settings.vre.ch_31_endpoint
+      req.headers['Authorization'] = "Bearer #{get_token}"
+      req.headers['Content-Type'] = 'application/json'
+      req.body = format_payload_for_vre
+    end
+
+    response_body = JSON.parse(response.body)
+    return true if response_body['ErrorOccurred'] == false
+
+    raise VREError
+  rescue VREError => e
+    log_exception_to_sentry(
+      e,
+      {
+        error_message: response_body['ErrorMessage']
+      },
+      { team: 'vfs-ebenefits' }
+    )
+  end
+
+  # SavedClaims require regional_office to be defined
+  def regional_office
+    []
+  end
+
+  private
+
+  def veteran_va_file_number(user)
+    service = BGS::PeopleService.new(user)
```

Comment:
This is a great example of how to refactor the above logic into it's own connection service and subsequently call it in the model.